src/app/BreedProcessor.ts

```diff
@@ -144,4 +149,44 @@ export class BreedProcessor {
     );
     return min([herdMaxGrow, bankContains]) as number;
   }
+
+  private reduceGain(
+    predator: Predator,
+    animalsGain: [AnimalNames, number][],
+  ): void {
+    if (predator.theName === AnimalNames.WOLF) {
+      const horseIndex = this.getAnimalIndex(
+        animalsGain,
+        AnimalNames.HORSE,
+      );
+      if (horseIndex === -1) {
+        animalsGain.splice(0);
+      } else {
+        animalsGain.splice(0, horseIndex);
+        animalsGain.splice(1, animalsGain.length);
+      }
+    }
+    if (predator.theName === AnimalNames.FOX) {
+      const rabbitIndex = this.getAnimalIndex(
+        animalsGain,
+        AnimalNames.RABBIT,
+      );
+      if (rabbitIndex !== -1) {
+        animalsGain.splice(rabbitIndex, 1);
```

Comment:
Could you try to redo it, so it bases on the configuration file?

Reply:
done